shm/Island.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Island.hpp"
+
+
+Island::Island(Coordinates coordinates, Time* time)
+    : position_(coordinates) {
+         std::unique_ptr<Store> store(new Store(time));
+         store_ = std::move(store);
```

Comment:
Maybe use DI? it will be hard to test Island, because you can't mock the store here.